solr/core/src/java/org/apache/solr/handler/admin/SystemInfoHandler.java

```diff
@@ -339,7 +341,10 @@ public SimpleOrderedMap<Object> getSecurityInfo(SolrQueryRequest req) {
         RuleBasedAuthorizationPluginBase rbap = (RuleBasedAuthorizationPluginBase) auth;
         Set<String> roles = rbap.getUserRoles(req.getUserPrincipal());
         info.add("roles", roles);
-        info.add("permissions", rbap.getPermissionNamesForRoles(roles));
+        info.add(
+            "permissions",
+            rbap.getPermissionNamesForRoles(
+                Stream.concat(roles.stream(), Stream.of("*", null)).collect(Collectors.toSet())));
```

Comment:
I considered letting `getPermissionNamesForRoles` always consider `*` and `null` roles, since this is the only place it is used, but the name of the method suggests that you want permissions for exactly those roles, so I expanded on the caller side instead...

Reply:
I'm hitting a NPE with Solr 9.2 in this line. I believe the issue is that `roles` can actually be null, if the user has no roles defined. Unfortunately, this is the case for the `$` user, that we use in `PKIAuthenticationPlugin`, this this causes the `/solr/admin/info/system` API not to work when using `nodes` (which the UI uses to populate the Nodes tab).

Reply:
https://issues.apache.org/jira/browse/SOLR-16729

Reply:
Ugh, sorry, I created SOLR-16730